### PR TITLE
[WIP] Add load balancing based on host health status

### PR DESCRIPTION
**Pull request by gksinghjsr**

### CODE REVIEW

File: pkg/target/host.go [hunk 1]
```diff
@@ -127,18 +114,22 @@ func (h *Host) Stream(wg *sync.WaitGroup) {
...
+		// not retrying here, file descriptor may be lost
+		h.Lg.Error("error closing the connection",
+			zap.Error(err))
+	}
+	h.updateHostConnection(nil)
+	h.updateHostHealthStatus(updateHostHealthStatus, false)
+}
+
 // Write does the remote write, i.e. sends the data
 func (h *Host) Write(b []byte) (nn int, err error) {
```
Comment (avereha) [inline pkg/target/host.go:134]: Writing to the buffer can call Flush() when the buffer is full. Flush() will use the connection. 
We need to use the connection lock here.

Comment (grzkv) [reply pkg/target/host.go:134]: Also, I think the writer and connection locking is so interconnected that it probably makes sense to make a single RW mutex both for connection and writer. We can then lock it for writing when using the writer.

**Comment by grzkv** [comment]
Since there is an alternative implementation going on, I am postponing reviewing this. Is this ok, @gksinghjsr or should I review this first? Also, please make a WIP PR for the new version too.

**Comment by grzkv** [comment]
@gksinghjsr Should we close this?

**Pull request closed without merging**